### PR TITLE
Fixing some caching issues for AzureCosmosDBSemanticCache

### PR DESCRIPTION
**Pull request by aayush3011**
Fixing some issues for AzureCosmosDBSemanticCache
  - Added the entry for "AzureCosmosDBSemanticCache" which was missing in langchain/cache.py   
  - Added application name when creating the MongoClient for the AzureCosmosDBVectorSearch, for tracking purposes.

@baskaryan, can you please review this PR, we need this to go in asap. These are just small fixes which we found today in our testing.

### CODE REVIEW

**Comment by aayush3011** [comment]
@baskaryan , can you please re-run the pipelines, there were some lint issues. The builds should pass in this iteration.

**Comment by aayush3011** [comment]
@baskaryan , can you please re-run the pipelines.

**Comment by aayush3011** [comment]
@baskaryan, can we go ahead merge the PR, once you get a chance.

File: libs/community/langchain_community/cache.py [hunk 1]
```diff
@@ -1868,6 +1869,7 @@ def __init__(
         ef_construction: int = 64,
         ef_search: int = 40,
         score_threshold: Optional[float] = None,
+        application_name: str,
```
Comment (baskaryan) [inline libs/community/langchain_community/cache.py:1872]: we should assign the default value here

Comment (aayush3011) [reply libs/community/langchain_community/cache.py]: Updated the code.

**Comment by aayush3011** [comment]
@baskaryan , can you please re-run the pipelines, I made changes requested as per the comments.

**Comment by aayush3011** [comment]
@baskaryan, can you take a final look at the PR, and we can merge it.

File: libs/community/langchain_community/vectorstores/azure_cosmos_db.py [hunk 2]
```diff
@@ -119,6 +120,7 @@ def from_connection_string(
         connection_string: str,
         namespace: str,
         embedding: Embeddings,
+        application_name: str,
```
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/azure_cosmos_db.py:123]: same comment as in other file

Comment (aayush3011) [reply libs/community/langchain_community/vectorstores/azure_cosmos_db.py]: Updated the code

**Comment by aayush3011** [comment]
@baskaryan, can you take a final look at the PR, and we can merge it.

**Comment by aayush3011** [comment]
@baskaryan, can you please do a final review, and merge the PR.

**Comment by aayush3011** [comment]
@baskaryan, any update on this?

**Comment by aayush3011** [comment]
@baskaryan, update on this PR. Can we go ahead and merge these changes.

**Comment by aayush3011** [comment]
@baskaryan, any update on this PR.

**Review by baskaryan — APPROVED** [review]

**Pull request merged**